### ISSUE TITLE
Meta batches (for training with joint loss) & full refactoring of data.py

### ISSUE THREAD

**Issue by s1m0000n** [s1m0000n/multitask-transformers#35]
First experiments produce better results with loss as sum of losses, seems quite promissing

**Comment by s1m0000n** [comment]
https://colab.research.google.com/drive/1q0Ob1eOmQSaja2cHWFoPwN28dO0id38K?usp=sharing

**Comment by s1m0000n** [comment]
This is the normal way of dealing with epochs:
```python
for epoch in range(num_epochs):
    for batch in train_dataloader:
```
Not this!
```python
if i == num_training_steps:
        break
```

Possible solution - make `__iter__()` return exactly iterable for data in one epoch
